### PR TITLE
feat(clients): update README with documentation, usage and more

### PR DESCRIPTION
**Pull request by trivikr**
*Issue #, if available:*
Fixes: https://github.com/aws/aws-sdk-js-v3/issues/1878
Fixes: https://github.com/aws/aws-sdk-js-v3/issues/1905

*Description of changes:*
Updates client README with the following:
- Documentation of the client.
- Installation.
- Getting Started: Importing and Usage examples.
- Troubleshooting.
- Getting Help.
- Contributing.
- License.

Example [README for DynamoDB Client](https://github.com/trivikr/aws-sdk-js-v3/blob/update-client-readme/clients/client-dynamodb/README.md)

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: clients/client-accessanalyzer/README.md [hunk 1]
```diff
@@ -3,4 +3,203 @@
 [![NPM version](https://img.shields.io/npm/v/@aws-sdk/client-accessanalyzer/latest.svg)](https://www.npmjs.com/package/@aws-sdk/client-accessanalyzer)
 [![NPM downloads](https://img.shields.io/npm/dm/@aws-sdk/client-accessanalyzer.svg)](https://www.npmjs.com/package/@aws-sdk/client-accessanalyzer)
 
-For SDK usage, please step to [SDK readme](https://github.com/aws/aws-sdk-js-v3).
+## Description
+
+AWS SDK for JavaScript AccessAnalyzer Client for Node.js, Browser and React Native.
```
Comment (AllanZhengYP) [inline clients/client-accessanalyzer/README.md:8]: Should we add and asterisk here stating that whether the clients runs on browser depends on the CORS setting on serverside?

File: clients/client-accessanalyzer/README.md [hunk 2]
```diff
@@ -3,4 +3,203 @@
...
+  .catch((error) => {
+    // error handling.
+  });
+
+// callbacks.
+client.createAnalyzer(params, (err, data) => {
+  // proccess err and data.
+});
+```
+
```
Comment (AllanZhengYP) [inline clients/client-accessanalyzer/README.md:159]: Maybe add a line here stating that 
```console
This style may results to bigger bundle size, and may be dropped in next major version.
```

File: codegen/smithy-aws-typescript-codegen/src/main/resources/software/amazon/smithy/aws/typescript/codegen/README.md.template [hunk 3]
```diff
@@ -3,4 +3,193 @@
...
+## Getting Help
+
+Please use these community resources for getting help.
+We use the GitHub issues for tracking bugs and feature requests, but have limited bandwidth to address them.
+
+- Visit [Developer Guide](https://docs.aws.amazon.com/sdk-for-javascript/v3/developer-guide/welcome.html)
+  or [API Reference](https://docs.aws.amazon.com/AWSJavaScriptSDK/v3/latest/index.html).
+- Ask a question on [StackOverflow](https://stackoverflow.com/questions/tagged/aws-sdk-js) and tag it with `aws-sdk-js`.
+- Join the AWS JavaScript community on [gitter](https://gitter.im/aws/aws-sdk-js-v3).
+- If it turns out that you may have found a bug, please [open an issue](https://github.com/aws/aws-sdk-js-v3/issues/new/choose).
```
Comment (AllanZhengYP) [inline codegen/smithy-aws-typescript-codegen/src/main/resources/software/amazon/smithy/aws/typescript/codegen/README.md.template:181]: Adding a link to all of our blog post will be awesome!

```suggestion
- Visit [Developer Guide](https://docs.aws.amazon.com/sdk-for-javascript/v3/developer-guide/welcome.html)
  or [API Reference](https://docs.aws.amazon.com/AWSJavaScriptSDK/v3/latest/index.html).
- Find if we already have [a deep-dive blog post](https://aws.amazon.com/blogs/developer/category/developer-tools/aws-sdk-for-javascript-in-node-js/) on this topic.
- Ask a question on [StackOverflow](https://stackoverflow.com/questions/tagged/aws-sdk-js) and tag it with `aws-sdk-js`.
- Join the AWS JavaScript community on [gitter](https://gitter.im/aws/aws-sdk-js-v3).
- If it turns out that you may have found a bug, please [open an issue](https://github.com/aws/aws-sdk-js-v3/issues/new/choose).
```

**Review by AllanZhengYP — APPROVED** [review]

File: clients/client-accessanalyzer/README.md (hunk 1 above)
Comment (trivikr) [reply clients/client-accessanalyzer/README.md:8]: Created a follow-up task to be taken after `v3.3.0` release https://github.com/aws/aws-sdk-js-v3/issues/1913

File: clients/client-accessanalyzer/README.md (hunk 2 above)
Comment (trivikr) [reply clients/client-accessanalyzer/README.md:161]: Added in [`6967168` (#1907)](https://github.com/aws/aws-sdk-js-v3/pull/1907/commits/6967168e626185af460d1c7a75d235ea42b8ce95)

File: codegen/smithy-aws-typescript-codegen/src/main/resources/software/amazon/smithy/aws/typescript/codegen/README.md.template (hunk 3 above)
Comment (trivikr) [reply codegen/smithy-aws-typescript-codegen/src/main/resources/software/amazon/smithy/aws/typescript/codegen/README.md.template]: Thanks for the suggestion! Added in [`68a54bc` (#1907)](https://github.com/aws/aws-sdk-js-v3/pull/1907/commits/68a54bc6b25954c565eeff617d9a06dc2e136bbb)

**Pull request merged**